corva/settings.py

```diff
@@ -1,15 +1,40 @@
 from os import getenv
-from typing import Final
+from typing import Optional
 
-API_ROOT_URL: Final[str] = getenv('API_ROOT_URL')
-DATA_API_ROOT_URL: Final[str] = getenv('DATA_API_ROOT_URL')
-APP_KEY: Final[str] = getenv('APP_KEY')
-APP_NAME: Final[str] = getenv('APP_NAME')
-API_KEY: Final[str] = getenv('API_KEY')
+from pydantic import BaseSettings
 
-# Logger
-LOG_LEVEL: Final[str] = getenv('LOG_LEVEL', 'WARN')
-LOG_ASSET_ID: Final[int] = int(getenv('LOG_ASSET_ID', -1))
 
-# Storage
-CACHE_URL: Final[str] = getenv('CACHE_URL')
+class CorvaSettings(BaseSettings):
+    # api
+    API_ROOT_URL: Optional[str] = None
+    DATA_API_ROOT_URL: Optional[str] = None
+    API_KEY: Optional[str] = None
+
+    # cache
+    CACHE_URL: Optional[str] = None
+
+    # logger
+    LOG_LEVEL: str = 'WARN'
+
+    # misc
+    APP_KEY: Optional[str] = None  # <provider>.<app-name-with-dashes>
+
+    @property
+    def APP_NAME(self) -> str:
+        if app_name := getenv('APP_NAME') is not None:
+            return app_name
+
+        app_name_with_dashes = self.APP_KEY.split('.')[1]
+        app_name = app_name_with_dashes.replace('-', ' ').title()
+
+        return app_name
+
+    @property
+    def PROVIDER(self) -> str:
+        if provider := getenv('PROVIDER') is not None:
+            return provider
+
+        return self.APP_KEY.split('.')[0]
+
+
+CORVA_SETTINGS = CorvaSettings()
```

Comment:
Can we call this SETTINGS instead of CORVA_SETTINGS?

Reply:
Named it `CORVA_SETTINGS` to avoid vague imports like: `from corva.settings import SETTINGS`, as pointed out in some previous PR

Reply:
Yes, renamed the file, class and global variable